util/udt_util.cc

```diff
@@ -230,28 +233,29 @@ Status TimestampRecoveryHandler::ReconcileTimestampDiscrepancy(
 }
 
 Status HandleWriteBatchTimestampSizeDifference(
+    const WriteBatch* batch,
     const std::unordered_map<uint32_t, size_t>& running_ts_sz,
     const std::unordered_map<uint32_t, size_t>& record_ts_sz,
     TimestampSizeConsistencyMode check_mode,
-    std::unique_ptr<WriteBatch>& batch) {
+    std::unique_ptr<WriteBatch>* new_batch) {
   // Quick path to bypass checking the WriteBatch.
   if (AllRunningColumnFamiliesConsistent(running_ts_sz, record_ts_sz)) {
     return Status::OK();
   }
   bool need_recovery = false;
   Status status = CheckWriteBatchTimestampSizeConsistency(
-      batch.get(), running_ts_sz, record_ts_sz, check_mode, &need_recovery);
+      batch, running_ts_sz, record_ts_sz, check_mode, &need_recovery);
   if (!status.ok()) {
     return status;
-  } else if (need_recovery) {
-    SequenceNumber sequence = WriteBatchInternal::Sequence(batch.get());
+  } else if (need_recovery && new_batch != nullptr) {
```

Comment:
Feels like the non-nullness of `new_batch` should be an assertion (not a check) when `need_recovery` is true. I think `need_recovery==true` implies `kReconcileInconsistency`, in which case the caller not passing a `new_batch` is a logic error, right?

Reply:
Agreed, the caller asks for `kReconcileInconsistency` but didn't provide a `new_batch` in case actual reconciliation happens. That's an irrational caller that should be called out.